Documentation/LibraryExtensibleRegexLiterals.md

```diff
@@ -32,66 +61,64 @@ let __C3 = builder.concatenate(__A3, __C2)
 return builder.finalize(__C3)
 ```
 
-Since regex syntax is acyclic, we don't strictly need tokens for terms, but it keeps it cleaner in my mind. I didn't assign types to captures in this example...
+This doesn't demonstrate typed captures, however...
 
-Note: enumeration of character ranges only really works for single-scalar grapheme clusters and would be ordered based on scalar value. For scalars that have decompositions.... well... maybe this only makes sense for ASCII. Also, what to do about CR-LF?
+Of course, we can also provide a built-in facility to pretty-print it back out, which would be useful for libraries wrapping string-based engines.
 
+Note: Enumeration of character ranges only really works for single-scalar grapheme clusters and would be ordered based on scalar value. It would also probably only make sense for single-segment normalization-invariant scalars. It's also essentially meaningless unless someone is really doing data processing and wants the scalar value range behavior. Also, what to do about CR-LF?
 
-## Supported features problem
+### Supported features problem
 
 There needs to be a way to surface the supported feature set:
 
 1) This conformer doesn't support a particular feature that's used
 2) This call site or usage doesn't support a particular feature that's used
 
-### 1) Conformer feature set
+This approach addresses concern #1. The compiler statically parses the regex, looking to invoke corresponding function declarations in some scope determined by the conforming type. If a declaration is not found, compiler issues an unsupported feature error. Libraries put availability on the declarations, which statically communicates their feature set and is the means for adding future functionality.
 
-Every single "feature" like a character class or some other meta-thingy, looks for a corresponding function definition on a conforming type. That is, we parse the regex, even providing intended semantics, while the conformer implements this. If the conformer doesn't provide a function definition, we generate a compilation error. Thus, conformers encode their feature set through ad-hoc function declarations, just like custom string interpolations.
+Concern #2 is... **TBD**. One example is if you're trying to run with grapheme cluster semantics, scalar properties aren't available (at least, beyond the subset that Swift can meaningfully prescribe grapheme cluster semantics for). APIs probably need some way to enforce this statically (and/or dynamically with traps).
```

Comment:
Yup, the whole Unicode-is-universal or Unicode-is-algebraic thing kinda falls apart pretty quickly. The stdlib can provide some case conversion and case insensitivity, but we're starting to more formally delineate it as something that can only provide universal semantics and doesn't incorporate locale or large data such as language dictionaries for word breaking. Case conversion can change the count of a `String` for `ß`. But, we very much want to allow a higher level linguistic framework to use regex literals (with their specified feature subset) to do better matching.

We might go so far as to exclude anything that isn't ASCII from a regex literal character class range for the stdlib's conformance. Stdlib ordering is not particularly meaningful outside of use for programmer invariants. But you could imagine a framework conformer that does incorporate both locale and application domain (e.g. is this a German phonebook or a German dictionary?... because it actually matters ....).